src/widgets/answers/answers.tsx

```diff
@@ -0,0 +1,179 @@
+/** @jsx h */
+
+import { h, render } from 'preact';
+import cx from 'classnames';
+import { WidgetFactory, Template, Hit, Renderer } from '../../types';
+import defaultTemplates from './defaultTemplates';
+import {
+  createDocumentationMessageGenerator,
+  getContainerNode,
+  prepareTemplateProps,
+} from '../../lib/utils';
+import { component } from '../../lib/suit';
+import Answers from '../../components/Answers/Answers';
+import connectAnswers, {
+  AnswersRendererOptions,
+  AnswersConnectorParams,
+} from '../../connectors/answers/connectAnswers';
+
+const withUsage = createDocumentationMessageGenerator({ name: 'answers' });
+const suit = component('Answers');
+
+const renderer = ({
+  renderState,
+  cssClasses,
+  containerNode,
+  templates,
+}): Renderer<AnswersRendererOptions, Partial<AnswersWidgetParams>> => (
+  { hits, isLoading, instantSearchInstance },
+  isFirstRendering
+) => {
+  if (isFirstRendering) {
+    renderState.templateProps = prepareTemplateProps({
+      defaultTemplates,
+      templatesConfig: instantSearchInstance.templatesConfig,
+      templates,
+    });
+    return;
+  }
+
+  render(
+    <Answers
+      cssClasses={cssClasses}
+      hits={hits}
+      isLoading={isLoading}
+      templateProps={renderState.templateProps}
+    />,
+    containerNode
+  );
+};
+
+export type AnswersTemplates = {
+  /**
+   * Template to use for the header. This template will receive an object containing `hits` and `isLoading`.
+   */
+  header?: Template<{
+    hits: Hit[];
+    isLoading: boolean;
+  }>;
+
+  /**
+   * Template to use for the loader.
+   */
+  loader?: Template;
+
+  /**
+   * Template to use for each result. This template will receive an object containing a single record.
+   */
+  item?: Template<Hit>;
+};
+
+export type AnswersCSSClasses = {
```

Comment:
We can use the `Partial` type to make these types more composable and reusable.

Reply:
https://github.com/algolia/instantsearch.js/pull/4581/commits/e8275509afe232d9c6d740a71b2114000896ce73